spec/relation_extension_spec.rb

```diff
@@ -29,9 +28,34 @@
       subject
     end
 
-    it 'invokes `Readyset.drop_cache!` with the parameterized query string that the relation ' \
-        'represents' do
+    it "invokes `Readyset.drop_cache!` with the relation's query string" do
       expect(Readyset).to have_received(:drop_cache!).with(sql: query_string)
     end
   end
+
+  describe '#readyset_explain' do
+    it "invokes `Readyset.readyset_explain` with the relation's query string" do
+      query = Cat.where(id: 1)
+      query_string = query.to_sql
+      allow(Readyset).to receive(:explain).with(query_string).
+        and_return(instance_double(Readyset::Explain))
+
+      query.readyset_explain
+
+      expect(Readyset).to have_received(:explain).with(query_string)
+    end
+
+    it 'returns the expected explain information' do
+      query = Cat.where(id: 1)
+      query_string = query.to_sql
+      explain = Readyset::Explain.new(id: 'q_0000000000000000',
+                                      text: 'SELECT * FROM "cats" WHERE ("id" = $1)',
+                                      supported: :yes)
+      allow(Readyset).to receive(:explain).with(query_string).and_return(explain)
+
+      output = query.readyset_explain
+
+      expect(output).to eq(explain)
+    end
+  end
```

Comment:
This looks a bit similar to `ready_set_spec`, but nested one-level deeper. 

What was the use-case for `relation_extension` having its own interface for `explain`?

Reply:
The idea is that users can just call `.readyset_explain` on an arbitrary ActiveRecord expression like `User.where(active: true).readyset_explain`. Without this, users would need to do something like `Readyset.explain(User.where(active: true))`. It's not a huge difference, but I think the former is more ergonomic, and it more closely matches [the syntax for the regular `EXPLAIN` command](https://guides.rubyonrails.org/active_record_querying.html#running-explain)